### PR TITLE
fix: Optimize export metric behavior

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Optimize export metric behavior in pr https://github.com/GreptimeTeam/greptimedb/pull/2928

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Avoid network when standalone and frontend are feeding themselves by `ExportMetricHandler`
- Default export metric schema be set to `information_schema` after solve https://github.com/GreptimeTeam/greptimedb/issues/2976

About discuss on https://github.com/GreptimeTeam/greptimedb/issues/2972

Because `GreptimeDBTelemetryTask` in `metasrv` is a bit complicate, so we’ll discuss it later

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)

Closes https://github.com/GreptimeTeam/greptimedb/issues/2975

### CODE REVIEW

File: src/servers/src/export_metrics.rs [hunk 1]
```diff
@@ -107,73 +110,127 @@ impl ExportMetricsTask {
...
+                self.config.db.clone(),
+                h,
+                filter,
+                interval,
+            ))
+        } else {
+            common_runtime::spawn_bg(write_system_metric_by_network(
+                self.headers.clone(),
+                format!(
+                    "http://{}/v1/prometheus/write?db={}",
```
Comment (killme2008) [inline src/servers/src/export_metrics.rs:152]: We can make the endpoint a completely valid URL, such as `http://x.x.x.x:4000/v/1/prometheus/write".

The benefits of this change:
1. We can support different protocols ,such as `https`(for example, use our cloud endpoints).
2. Supports other storages that compatible with promethues remote write.

**Review by killme2008 — APPROVED** [review]
Almostly LGTM

**Comment by killme2008** [comment]
@v0y4g3r @fengjiachun PTAL

**Comment by zhongzc** [comment]
I wholeheartedly agree with the configuration mentioned by @killme2008. Frankly speaking, it would be hard for me to hit the Approve button if we didn't follow through with this approach. 🥲

Here are my opinions:
https://github.com/GreptimeTeam/greptimedb/pull/2928#discussion_r1433634330
https://github.com/GreptimeTeam/greptimedb/pull/2928#discussion_r1433605689

**Comment by Taylor-lagrange** [comment]
In my original idea, I did retain the design of using the entire url as a parameter, for greater compatibility. In the subsequent review, the parameters are split to `endpoint` and `db` according to the advice. So we can reach an agreement to merge `endpoint` and `db` into one parameter (e.g. `http://127.0.0.1/v1/prometheus/write?db=information_schema`), which will have wider compatibility?

But as for distinguishing the two import methods (i.e. `remote_write`  for url and `self_import` for internal), I personally think it may increase users' understanding. Because only `frontend` and `standalone` can use `self_import` in certain case, others have to rely on url, because there is no good way to know the frontend address in the cluster.

How about yours opinion? @killme2008  @zhongzc

**Comment by killme2008** [comment]
> In my original idea, I did retain the design of using the entire url as a parameter, for greater compatibility. In the subsequent review, the parameters are split to `endpoint` and `db` according to the advice. So we can reach an agreement to merge `endpoint` and `db` into one parameter (e.g. `http://127.0.0.1/v1/prometheus/write?db=information_schema`), which will have wider compatibility?
> 
> But as for distinguishing the two import methods (i.e. `remote_write` for url and `self_import` for internal), I personally think it may increase users' understanding. Because only `frontend` and `standalone` can use `self_import` in certain case, others have to rely on url, because there is no good way to know the frontend address in the cluster.
> 
> How about yours opinion? @killme2008 @zhongzc

Yes, totally agree with the URL as configured parameter instead of `endpoint` and `db`.

Considering that `standalone` would be the most installed version, i think it's valuable for user to use`self_import`, and we can make it default for standalone.

File: config/standalone.example.toml [hunk 2]
```diff
@@ -230,11 +230,8 @@ parallel_scan_channel_size = 32
...
-# endpoint = "127.0.0.1:4000"
-# The database name of exported metrics stores, user needs to specify a valid database
-# db = ""
 # The interval of export metrics
 # write_interval = "30s"
-# HTTP headers of Prometheus remote-write carry
-# headers = {}
+# for `standalone`, `self_import` is recommend to collect metrics generated by itself
+# [export_metrics.self_import]
+# db = "information_schema"
```
Comment (zhongzc) [inline config/standalone.example.toml:237]: require a new line

**Review by zhongzc — APPROVED** [review]
LGTM

**Pull request merged**